src/go/pt-k8s-pxc-recovery/recover/recover.go

```diff
@@ -0,0 +1,297 @@
+	}
+	_, err := kubectl.RunCmd(c.Namespace, args...)
+	return err
+}
+
+func (c *Cluster) RestartPods() error {
+	for i := 0; i < c.Size; i++ {
+		args := []string{
+			"delete",
+			"pod",
+			c.Name + "-pxc-" + strconv.Itoa(i),
+			"--force",
+			"--grace-period=0",
+		}
+		_, err := kubectl.RunCmd(c.Namespace, args...)
+		if err != nil && !strings.Contains(err.Error(), "pods") && !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) CheckPodReady(podID int) (bool, error) {
+	args := []string{
+		"get",
+		"pod", c.Name + "-pxc-" + strconv.Itoa(podID),
+		"-o",
+		"jsonpath='{.status.containerStatuses[0].ready}'",
+	}
+	output, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return false, err
+	}
+	return strings.Trim(output, "'") == "true", nil
+}
+
+func (c *Cluster) PodZeroReady() error {
+	podZeroStatus := false
+	var err error
+	for !podZeroStatus {
+		time.Sleep(time.Second * 10)
+		podZeroStatus, err = c.CheckPodReady(0)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) CheckPodPhase(podID int, phase string) (bool, error) {
+	args := []string{
+		"get",
+		"pod", c.Name + "-pxc-" + strconv.Itoa(podID),
+		"-o",
+		"jsonpath='{.status.phase}'",
+	}
+	output, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return false, err
+	}
+	return strings.Trim(output, "'") == phase, nil
+}
+
+func (c *Cluster) AllPodsRunning() error {
+	for i := 0; i < c.Size; i++ {
+		running := false
+		var err error
+		for !running {
+			time.Sleep(time.Second * 10)
+			running, err = c.CheckPodPhase(i, "Running")
+			if err != nil && !strings.Contains(err.Error(), "NotFound") {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) RunCommandInPod(podID int, cmd ...string) (string, error) {
+	args := []string{
+		"exec",
+		c.Name + "-pxc-" + strconv.Itoa(podID),
+		"--",
+	}
+	args = append(args, cmd...)
+	output, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return "", err
+	}
+	return output, nil
+}
+
+func (c *Cluster) SetSSTInProgress() error {
+	for i := 0; i < c.Size; i++ {
+		_, err := c.RunCommandInPod(i, "touch", "/var/lib/mysql/sst_in_progress")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) AllPodsReady() error {
+	for i := 0; i < c.Size; i++ {
+		podReadyStatus := false
+		var err error
+		for !podReadyStatus {
+			time.Sleep(time.Second * 10)
+			podReadyStatus, err = c.CheckPodReady(i)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) FindMostRecentPod() error {
+	var podID int
+	seqNo := 0
+	re := regexp.MustCompile(`(?m)seqno:\s*(\d*)`)
+	for i := 0; i < c.Size; i++ {
+		output, err := c.RunCommandInPod(i, "cat", "/var/lib/mysql/grastate.dat")
+		if err != nil {
+			return err
+		}
+		match := re.FindStringSubmatch(output)
+		if len(match) < 2 {
+			return fmt.Errorf("unable to get seqno")
```

Comment:
This is a good example of the necessity to provide more context to the errors. Imagine error here. A user would see
```
...
Finding the most recent pod
Error: unable to get seqno
```
how useful is information in this error?

what the heck is seqno? on what pod it happend? why did it happen?